### ISSUE TITLE
2.7.0 Introduced Breaking Changes

### ISSUE THREAD

**Issue by chulbert** [Swinject/Swinject#434]
Swinject 2.7.0 introduced breaking API changes, specifically the public protocol `ServiceKeyOption` added method `func hash(into: inout Hasher)`.  There may be others.  While I do not see a formal statement on the project's versioning policies, it's standard practice to follow so-called "semantic versioning" and, as such, breaking API changes should require a major version increment (3.0).

Please consider going forward.

**Comment by jakubvano** [comment]
You are correct, we should strive for the semantic versioning - breaking the `ServiceKeyOption` was an oversight 😞 
Apologies for any inconveniences.

**Comment by PeqNP** [comment]
Sorry if this is obvious, but for those who need a "workaround" until this is fixed, you can pin your version of Swinject to v2.6.2.

**Comment by 1ucas** [comment]
Hello,
considering the time Swinject 2.7.0 have been around, should we still consider a version 3 as the next minor release?

If not, should we close this issue?

**Comment by chulbert** [comment]
I’m not sure I understand the question.  The “next release” should in my opinion increment from the current release according to semantic versioning.

**Comment by 1ucas** [comment]
Is there anything you believe we should do to solve this issue?

**Comment by chulbert** [comment]
I suppose a test in the build suite to guard against this situation in the future might be a reasonable step. Otherwise I don’t think there are many direct steps to take short of a major release.  Go ahead and close and promise not to let it happen again.  ;)

**Comment by 1ucas** [comment]
I'm adding some constraints to our CD workflow. 

I'll keep this issue open and link it on the PR. 

Thanks!
